packages/dev/core/src/FlowGraph/Blocks/Event/flowGraphMeshPickEventBlock.ts

```diff
@@ -53,19 +55,27 @@ export class FlowGraphMeshPickEventBlock extends FlowGraphEventBlock {
      * @internal
      */
     public _cancelPendingTasks(context: FlowGraphContext): void {
-        const mesh = context.getVariable(this._meshVariableName) as AbstractMesh;
+        const mesh = context._getExecutionVariable(this, "mesh");
         const pickObserver = context._getExecutionVariable(this, "meshPickObserver");
         const disposeObserver = context._getExecutionVariable(this, "meshDisposeObserver");
 
         mesh.getScene().onPointerObservable.remove(pickObserver);
         mesh.onDisposeObservable.remove(disposeObserver);
 
+        context._deleteExecutionVariable(this, "mesh");
         context._deleteExecutionVariable(this, "meshPickObserver");
         context._deleteExecutionVariable(this, "meshDisposeObserver");
     }
 
     public getClassName(): string {
         return "FGMeshPickEventBlock";
     }
+
+    public serialize(serializationObject?: any): void {

```

Comment:
a thought about the serialize api structure. if you do something like this:

```
public serialize(serializationObject?: any = {}): void {
// ....
return serializationObject;
}
```

You can avoid creating an object before calling the function, and this unction could look like this:

```
        super.serialize(serializationObject);
        serializationObject.config.path = this.config.path.serialize(;
```